host/libcontainer_backend.go

```diff
@@ -602,85 +629,60 @@ func (l *LibvirtLXCBackend) Run(job *host.Job, runConfig *RunConfig, rateLimitBu
 		return err
 	}
 
-	domain := &lt.Domain{
-		Type:   "lxc",
-		Name:   job.ID,
-		Memory: lt.UnitInt{Value: 1, Unit: "GiB"},
-		OS: lt.OS{
-			Type: lt.OSType{Value: "exe"},
-			Init: "/.containerinit",
-		},
-		Devices: lt.Devices{
-			Filesystems: []lt.Filesystem{
-				{
-					Type:   "mount",
-					Source: lt.FSRef{Dir: rootPath},
-					Target: lt.FSRef{Dir: "/"},
-				},
-				{
-					Type:   "ram",
-					Source: lt.FSRef{Usage: "65535"}, // 64MiB
-					Target: lt.FSRef{Dir: "/dev/shm"},
-				},
+	if !job.Config.HostNetwork {
+		config.Hostname = hostname
+		config.Namespaces = append(config.Namespaces, configs.Namespace{Type: configs.NEWNET})
+		config.Networks = []*configs.Network{
+			{
+				Type:    "loopback",
+				Address: "127.0.0.1/0",
+				Gateway: "localhost",
 			},
-			Consoles: []lt.Console{{Type: "pty"}},
-		},
-		Resource: &lt.Resource{
-			Partition: "/machine/" + job.Partition,
-		},
-		OnPoweroff: "preserve",
-		OnCrash:    "preserve",
+			{
+				Type:              "veth",
+				Name:              "eth0",
+				Bridge:            l.bridgeName,
+				Address:           initConfig.IP,
+				Gateway:           initConfig.Gateway,
+				Mtu:               1500,
+				HostInterfaceName: "veth" + random.String(4),
```

Comment:
Ah yes.

So after digging around, there is no code in libcontainer / runc which sets `HostInterfaceName` (which is why I just went with this random name). Looking at Docker v.1.11.1, it actually uses [libnetwork's GenerateIfaceName](https://github.com/docker/docker/blob/v1.11.1/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go#L889-L893) to do this, which just [tries a few times](https://github.com/docker/docker/blob/v1.11.1/vendor/src/github.com/docker/libnetwork/netutils/utils_linux.go#L33-L50).

We already have libnetwork vendored, so we could use that same method, or we could use an incremental int (although we will need to sychronise that across an udpate / restart). WDYT?